browser/src/control/jsdialog/Widget.ColorPicker.ts

```diff
@@ -103,49 +103,266 @@ function createColor(
 	color.setAttribute('index', index);
 	color.tabIndex = 0;
 	if (themeData) color.setAttribute('theme', themeData);
 
 	color.innerHTML = isCurrent ? '&#149;' : '&#160;';
 
-	color.addEventListener('click', (event: MouseEvent) => {
-		const target = event.target as Element;
-		const colorCode = target.getAttribute('name');
-		const themeData = target.getAttribute('theme');
-
-		if (colorCode != null) {
-			if (colorCode) {
-				builder._sendColorCommand(builder, widgetData, colorCode, themeData);
-				builder.callback(
-					'colorpicker',
-					'hidedropdown',
-					widgetData,
-					themeData ? themeData : colorCode,
+	color.addEventListener('click', (event: MouseEvent) =>
+		handleColorSelection(
+			event.target as HTMLElement,
+			builder,
+			widgetData,
+			palette,
+		),
+	);
+
+	// Handle keyboard navigation and selection
+	color.addEventListener('keydown', (event: KeyboardEvent) => {
+		switch (event.key) {
+			case 'Enter':
+			case ' ':
+				// Space or Enter key selects the color
+				handleColorSelection(
+					event.target as HTMLElement,
 					builder,
-				);
-			} else {
-				builder._sendColorCommand(builder, widgetData, 'transparent');
-				builder.callback(
-					'colorpicker',
-					'hidedropdown',
 					widgetData,
-					'transparent',
-					builder,
+					palette,
+				);
+				event.preventDefault();
+				break;
+			case 'ArrowRight':
+				moveFocus(color, 'next', 'horizontal');
+				event.preventDefault();
+				break;
+			case 'ArrowLeft':
+				moveFocus(color, 'previous', 'horizontal');
+				event.preventDefault();
+				break;
+			case 'ArrowDown':
+				moveFocus(
+					color,
+					'next',
+					'vertical',
+					parentContainer.nextElementSibling,
 				);
+				event.preventDefault();
+				break;
+			case 'ArrowUp':
+				moveFocus(
+					color,
+					'previous',
+					'vertical',
+					parentContainer.previousElementSibling,
+				);
+				event.preventDefault();
+				break;
+			default:
+				break;
+		}
+	});
+
+	return color;
+}
+
+function handleColorSelection(
+	target: HTMLElement,
+	builder: any,
+	widgetData: ColorPaletteWidgetData,
+	palette: ColorPalette,
+) {
+	const colorCode = target.getAttribute('name');
+	const themeData = target.getAttribute('theme');
+
+	if (colorCode != null) {
+		builder._sendColorCommand(builder, widgetData, colorCode, themeData);
+		builder.callback(
+			'colorpicker',
+			'hidedropdown',
+			widgetData,
+			themeData ? themeData : colorCode,
+			builder,
+		);
+	} else {
+		builder._sendColorCommand(builder, widgetData, 'transparent');
+		builder.callback(
+			'colorpicker',
+			'hidedropdown',
+			widgetData,
+			'transparent',
+			builder,
+		);
+	}
+	// Update the recent colors list
+	const recentRow = palette[palette.length - 1];
+	if (recentRow.indexOf(colorCode) !== -1)
+		recentRow.splice(recentRow.indexOf(colorCode), 1);
+	recentRow.unshift(colorCode);
+	window.prefs.set('recentColor', JSON.stringify(recentRow));
+}
+
+function moveFocus(
```

Comment:
this should be moved to some helper/util class which can be reused on any other popup, example: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/jsdialog/Util.FocusCycle.js

Please create new file there: eg. `control/jsdialog/Util.KeyboardGridNavigation.ts`